src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -378,11 +385,21 @@ def update_existing_check_in(
     if processing_errors:
         raise ProcessingErrorsException(processing_errors, monitor=monitor)
 
-    updated_checkin: _CheckinUpdateKwargs = {
-        "status": updated_status,
-        "duration": updated_duration,
-        "date_updated": start_time,
-    }
+    updated_checkin: _CheckinUpdateKwargs = {}
+
+    if updated_duration:
+        updated_checkin["duration"] = updated_duration
+
+    # When processing an out-of-order in-progress check-in we ONLY need to set
+    # the date_in_progress. We do NOT update the duration, since it's likely
+    # we want to actually calculate this
+    if is_out_of_order_in_progress:
+        updated_checkin["date_in_progress"] = start_time
+        existing_check_in.update(**updated_checkin)
```

Comment:
Do we always want to update the `date_in_progress`? Or should it only be if it's newer?